From 29f6ca1c5f4de56cb6e08f4b912b6618297ac013 Mon Sep 17 00:00:00 2001
From: Xiyue Deng <manphiz@gmail.com>
Date: Mon, 30 Sep 2024 23:45:29 -0700
Subject: [PATCH] Add emacs-editing-major-modes metapackage

* Recommends all Emacs major modes for editing available in Debian.
---
 debian/control | 61 ++++++++++++++++++++++++++++++++++++++++++++++++++
 1 file changed, 61 insertions(+)

diff --git a/debian/control b/debian/control
index 1d3f97f..5b176e7 100644
--- a/debian/control
+++ b/debian/control
@@ -41,3 +41,64 @@ Description: Miscellaneous add-ons for Emacs
  .
  See /usr/share/doc/emacs-goodies-el/README.Debian.gz for a discussion
  of the disposition of previously bundled add-ons.
+
+Package: emacs-editing-major-modes
+Architecture: all
+Depends: ${misc:Depends}
+Recommends:
+ elpa-agda2-mode,
+ elpa-apache-mode,
+ elpa-ats2-mode,
+ elpa-bazel-mode,
+ elpa-bison-mode,
+ elpa-bpftrace-mode,
+ elpa-caml,
+ elpa-clojure-mode,
+ elpa-cmake-mode,
+ elpa-csv-mode,
+ elpa-dap-mode,
+ elpa-dart-mode,
+ elpa-debian-el,
+ elpa-dockerfile-mode,
+ elpa-dpkg-dev-el,
+ elpa-elm-mode,
+ elpa-fountain-mode,
+ elpa-git-commit,
+ elpa-git-modes,
+ elpa-gitlab-ci-mode,
+ elpa-gnuplot-mode,
+ elpa-go-mode,
+ elpa-graphviz-dot-mode,
+ elpa-haskell-mode,
+ elpa-inform-mode,
+ elpa-jabber,
+ elpa-jinja2-mode,
+ elpa-js2-mode,
+ elpa-kivy-mode,
+ elpa-kotlin-mode,
+ elpa-lua-mode,
+ elpa-markdown-mode,
+ elpa-matlab-mode,
+ elpa-meson-mode,
+ elpa-muttrc-mode,
+ elpa-nginx-mode,
+ elpa-php-mode,
+ elpa-pip-requirements,
+ elpa-protobuf-mode,
+ elpa-puppet-mode,
+ elpa-qml-mode,
+ elpa-racket-mode,
+ elpa-rust-mode,
+ elpa-scala-mode,
+ elpa-sml-mode,
+ elpa-systemd,
+ elpa-vala-mode,
+ elpa-web-mode,
+ elpa-yaml-mode,
+ gettext-el,
+Description: All editing major modes for Emacs
+ This is a metapackage that recommends all Emacs major modes including
+ programming modes, configuration file modes, etc.  Installing this
+ package enables Emacs to handle most well-known file formats for
+ editing, while also providing the flexibility for a user to choose
+ which they do or do not need.
-- 
2.45.2

